src/pick.rs

```diff
@@ -57,6 +58,62 @@ pub fn pick<T>(randomness: [u8; 32], n: usize, mut data: Vec<T>) -> Vec<T> {
     data.split_off(data.len() - n)
 }
 
+#[derive(Clone)]
+pub struct WeightedElement<T: Clone> {
+    pub element: T,
+    pub weight: f64,
+}
+
+/// Picks 1 element from a given weighted list.
+///
+///
+/// ## Examples
+///
+/// Pick 1 hat out of 3 hats with different rarity :
+///
+/// ```
+/// use nois::{randomness_from_str, pick_one_from_weighted_list, WeightedElement };
+///
+/// let randomness = randomness_from_str("9e8e26615f51552aa3b18b6f0bcf0dae5afbe30321e8d7ea7fa51ebeb1d8fe62").unwrap();
+///
+///
+/// let data = vec![
+///             WeightedElement {
+///                 element: "green hat",
+///                 weight: 0.4,
+///             },
+///             WeightedElement {
+///                 element: "viking helmet",
+///                 weight: 0.55,
+///             },
+///             WeightedElement {
+///                 element: "rare gold crown",
+///                 weight: 0.05,
+///             },
+///         ];
+/// let picked = pick_one_from_weighted_list(randomness,  &data);
+///
+/// assert_eq!(picked, "green hat");
+/// ```
+pub fn pick_one_from_weighted_list<T: Clone>(
+    randomness: [u8; 32],
+    elements: &[WeightedElement<T>],
```

Comment:
When you make this a list of pairs, all you test code becomes much simpler:
```suggestion
    elements: &[(T, u32)],
```